### PR TITLE
added dataAttr prop on ComboSelect component

### PR DESCRIPTION
**Pull request by rajkovukovic**

### CODE REVIEW

File: Components/ComboSelect.jsx [hunk 1]
```diff
@@ -22,8 +22,8 @@ export default class ComboSelect extends Component {
 		};
 		this.borderActive = this.props.borderActive ? this.props.borderActive : '#40b4e5';
 
-		this.iconSelectActive = props.iconSelectActive !== false || props.iconSelectActive !== 'off' ? props.iconSelectActive : true;
-		this.iconSelectInactive = props.iconSelectInactive !== false || props.iconSelectInactive !== 'off' ? props.iconSelectInactive : true;
+		this.iconSelectActive = props.iconSelectActive === false || props.iconSelectActive === 'off' ? true : props.iconSelectActive;
+		this.iconSelectInactive = props.iconSelectInactive === false || props.iconSelectInactive === 'off' ? true : props.iconSelectInactive;
```
Comment (MilosRasic) [inline Components/ComboSelect.jsx]: This and the change in the line above are from #19.

File: Components/ComboSelect.jsx [hunk 2]
```diff
@@ -45,6 +45,29 @@ export default class ComboSelect extends Component {
 			selected: -1,
 			search: this.props.search && (this.props.search == 'on' || this.props.search == 'smart' || this.props.search == 'off') ? this.props.search : 'off'
 		}
+
+		this.processTransformFunctions();
+	}
+	
+	processTransformFunctions(newProps) {
```
Comment (MilosRasic) [inline Components/ComboSelect.jsx]: Can we rename this to something that makes it clear what the method actually does? Like processDataAttributes or something? processTransformFunctions doesn't tell us much.

File: Components/ComboSelect.jsx [hunk 3]
```diff
@@ -73,15 +96,17 @@ export default class ComboSelect extends Component {
 		window.removeEventListener('touchstart', this.globalMouseClick);
 		window.removeEventListener('wheel', this.globalWheel);
 		this.refs.select.removeEventListener('keydown', this.requiredSelectKeydown);
-		this.refs.select.addEventListener('focus', this.selectFocus);
-		this.refs.select.addEventListener('focusout', this.deSelectFocus);
+		this.refs.select.removeEventListener('focus', this.selectFocus);
+		this.refs.select.removeEventListener('focusout', this.deSelectFocus);
```
Comment (MilosRasic) [inline Components/ComboSelect.jsx]: good catch

File: Components/ComboSelect.jsx [hunk 4]
```diff
@@ -144,19 +170,21 @@ export default class ComboSelect extends Component {
 					target = target.parentElement;
 					i++;
 
-					if (target && target == this.refs.scroll) {
+					if (target && target.innerHTML == this.refs.scroll.innerHTML) {
```
Comment (MilosRasic) [inline Components/ComboSelect.jsx]: Same problem as in #19. This was changed in master but the PR wants to revert it. Please update.

File: Components/ComboSelect.jsx [hunk 5]
```diff
@@ -144,19 +170,21 @@ export default class ComboSelect extends Component {
...
 						outside = false;
 					}
 				}
 
 				if (outside) {
 					event.stopPropagation();
+					event.preventDefault();
```
Comment (MilosRasic) [inline Components/ComboSelect.jsx]: Same problem as in #19. This was changed in master but the PR wants to revert it. Please update.

File: Components/ComboSelect.jsx [hunk 6]
```diff
@@ -144,19 +170,21 @@ export default class ComboSelect extends Component {
...
 				}
 
 			} else if (this.refs.comboSelect.getElementsByClassName(specialClass)[0].scrollTop + event.deltaY <= 0) {
 
 				this.refs.scroll.scrollTop = 0;
 				event.stopPropagation();
+				event.preventDefault();
```
Comment (MilosRasic) [inline Components/ComboSelect.jsx]: Same problem as in #19. This was changed in master but the PR wants to revert it. Please update.

File: Components/ComboSelect.jsx [hunk 7]
```diff
@@ -187,7 +215,7 @@ export default class ComboSelect extends Component {
 			while (this.checkParentElement(target) && i < 10) {
 				target = target.parentElement;
 				i++;
-				if (target == this.refs.comboSelect) {
+				if (target.innerHTML == this.refs.comboSelect.innerHTML) {
```
Comment (MilosRasic) [inline Components/ComboSelect.jsx]: Same problem as in #19. This was changed in master but the PR wants to revert it. Please update.

File: Components/ComboSelect.jsx [hunk 8]
```diff
@@ -236,13 +264,17 @@ export default class ComboSelect extends Component {
...
-		var { data, type, onChange, search, value, onToggle, text, map, sort, iconSelectInactive, iconSelectActive, defaultText, scrollHeight, scrollMaxHeight, preferredDirection, ...other } = this.props;
+		var { data, type, onChange, search, value, onToggle, text, map, sort, iconSelectInactive, iconSelectActive, defaultText, scrollHeight, scrollMaxHeight, preferredDirection, dataAttr, ...other } = this.props;
 
 		if (this.state.value === 0 || (this.state.value && ((this.state.value instanceof Array && this.state.value.length > 0) || !(this.state.value instanceof Array)))) {
-
 			head = (<div onClick={() => this.toggleMenu()}>
-				<div className={(this.props.disabled ? " disabled " : "") + "combo-select-head"} ref="head">
-					{stateText ? stateText : this.defaultText}<i className={this.icon}></i>
+				<div
+					{...transformDataAttributes(this.dropDownHeaderDataTransformer, this.props)}
```
Comment (MilosRasic) [inline Components/ComboSelect.jsx]: I don't think we need to spread here. transformDataAttributes() creates a new object.

File: Components/ComboSelect.jsx [hunk 9]
```diff
@@ -252,8 +284,13 @@ export default class ComboSelect extends Component {
 		} else {
 
 			head = (<div onClick={() => this.toggleMenu()}>
-				<div className={(this.props.disabled ? " disabled " : "") + "combo-select-head"} ref="head">
-					{stateText ? stateText : this.defaultText}<i className={this.icon}></i>
+				<div
+					{...transformDataAttributes(this.dropDownHeaderDataTransformer, this.props)}
```
Comment (MilosRasic) [inline Components/ComboSelect.jsx]: I don't think we need to spread here. transformDataAttributes() creates a new object.

File: Components/ComboSelect.jsx [hunk 10]
```diff
@@ -283,7 +320,10 @@ export default class ComboSelect extends Component {
 
 				return (
 					<div key={i}>
-						<ComboSelectItem item={item} selected={this.findSelectedByKey(item, this.state.text, 'text')}
+						<ComboSelectItem
+							{...transformDataAttributes(this.listItemDataTransformer, item)}
```
Comment (MilosRasic) [inline Components/ComboSelect.jsx]: I don't think we need to spread here. transformDataAttributes() creates a new object.

File: Components/ComboSelect.jsx [hunk 11]
```diff
@@ -956,7 +996,11 @@ export default class ComboSelect extends Component {
 		let body = this._generateBody();
 
 		return (
-			<div ref="comboSelect" className="combo-select">
+			<div
+				{...transformDataAttributes(this.wrapperDataTransformer, this.props)}
```
Comment (MilosRasic) [inline Components/ComboSelect.jsx]: I don't think we need to spread here. transformDataAttributes() creates a new object.

**Comment by rajkovukovic** [comment]
Updates refactored and new PR created [#23](https://github.com/gogoair/react-combo-select/pull/23)

**Comment by rajkovukovic** [comment]
Updates refactored and new PR created [#23](https://github.com/gogoair/react-combo-select/pull/23)

**Pull request closed without merging**